### PR TITLE
feat/ publish to slack from nodejs

### PR DESCRIPTION
**Pull request by Le-Bit**

### CODE REVIEW

File: functions/gmail-notifier/index.js [hunk 1]
```diff
@@ -76,6 +77,18 @@ exports.initWatch = async (req, res) => {
   return res.send(`Watch initialized on gmail inbox: ${email}`)
 }
 
+exports.postMessageToSlack = async (text) => {
```
Comment (AcevedoR) [inline functions/gmail-notifier/index.js:80]: would be better to have test

File: functions/gmail-notifier/index.js [hunk 2]
```diff
@@ -84,26 +97,33 @@ exports.onNewMessage = async (message, context) => {
...
+    ) // TODO: foreach
+    const mailSubject = messageResponse.data.payload.headers.filter(
+      (e) => e.name == 'Subject'
+    )[0].value
+    const textPart = messageResponse.data.payload.parts.filter(
+      (e) => e.mimeType == 'text/plain'
+    )[0]
+    const bodyPlain = Buffer.from(textPart.body.data, 'base64').toString('utf8')
+
+    await this.postMessageToSlack(
```
Comment (AcevedoR) [inline functions/gmail-notifier/index.js:118]: is the current test checking we are sending 'formatted' data to postMessageToSlack function ?

File: functions/gmail-notifier/index.js [hunk 3]
```diff
@@ -76,6 +77,18 @@ exports.initWatch = async (req, res) => {
   return res.send(`Watch initialized on gmail inbox: ${email}`)
 }
 
+exports.postMessageToSlack = async (text) => {
+  const token = 'slack-bot-token'
+  const web = new WebClient(token)
+  const conversationID = 'C021BGBSMFY'
```
Comment (AcevedoR) [inline functions/gmail-notifier/index.js:83]: this should be in some configuration file

File: functions/gmail-notifier/index.js [hunk 4]
```diff
@@ -76,6 +77,18 @@ exports.initWatch = async (req, res) => {
...
+  try {
+    const res = await web.chat.postMessage({ channel: conversationID, text })
+  } catch (err) {
+    console.error(err)
```
Comment (AcevedoR) [inline functions/gmail-notifier/index.js:87]: is this alright in nodejs ? or should be throwing error ?

**Review by AcevedoR — CHANGES_REQUESTED** [review]

File: functions/gmail-notifier/index.js (hunk 2 above)
Comment (Le-Bit) [reply functions/gmail-notifier/index.js:118]: Formatted in the same way a real message would be, the markdown interpreted part of the test message is unintentional

File: functions/gmail-notifier/index.js (hunk 3 above)
Comment (Le-Bit) [reply functions/gmail-notifier/index.js:83]: Agreed, should be moved into GCP secret i think

File: functions/gmail-notifier/index.js (hunk 4 above)
Comment (Le-Bit) [reply functions/gmail-notifier/index.js:87]: It's alright imho, that's the canonic way to catch errors when using the await syntax, it's the equivalent to ".catch(" after a ".then(".

File: functions/gmail-notifier/index.js (hunk 1 above)
Comment (Le-Bit) [reply functions/gmail-notifier/index.js:80]: Would it be better to mock this function entirely ?
I do not see a good testing strategy if not mocking the slack API, which is basically all what this function call

**Review by lhauspie — CHANGES_REQUESTED** [review]

**Pull request merged**